src/VisualStudio/Core/Def/Implementation/DesignerAttribute/VisualStudioDesignerAttributeService.cs

```diff
@@ -0,0 +1,402 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.DesignerAttribute;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.ErrorReporting;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Remote;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.Designer.Interfaces;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Shell.Services;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.DesignerAttribute
+{
+    internal class VisualStudioDesignerAttributeService
+        : ForegroundThreadAffinitizedObject, IDesignerAttributeService, INewDesignerAttributeServiceCallback
+    {
+        private readonly IThreadingContext _threadingContext;
+        private readonly IServiceProvider _serviceProvider;
+        private readonly IForegroundNotificationService _notificationService;
+        private readonly VisualStudioWorkspaceImpl _workspace;
+        private readonly IAsynchronousOperationListener _listener;
+
+        // cache the update service for cps projects. Computed on demand (slow), but then cached for
+        // quick responses after that.
+        private readonly ConcurrentDictionary<ProjectId, IProjectItemDesignerTypeUpdateService> _cpsProjects
+            = new ConcurrentDictionary<ProjectId, IProjectItemDesignerTypeUpdateService>();
```

Comment:
Copied the code for this (including how it is initialized) from the previous impl.